src/plugins/maps_legacy/public/map/base_maps_visualization.js

```diff
@@ -95,9 +92,11 @@ export function BaseMapsVisualizationProvider(getKibanaMap, mapServiceSettings)
       const centerFromUIState = uiState.get('mapCenter');
       options.zoom = !isNaN(zoomFromUiState) ? zoomFromUiState : this.vis.params.mapZoom;
       options.center = centerFromUIState ? centerFromUIState : this.vis.params.mapCenter;
-      const services = { toastService };
 
-      this._kibanaMap = getKibanaMap(this._container, options, services);
+      const modules = await lazyLoadMapsLegacyModules();
+      this._leaflet = modules.L;
```

Comment:
nit, but generally speaking, instance-variables are values or instances of a type. Here we kind of "hack" them to maintain a reference to a module so it can be reused in sub-classes.


It does seem that this initial loading is in `async` methods anyway, so you could just load it async there too:

https://github.com/elastic/kibana/blob/37c444a70c184ce8dc0ab5f5eb8fe684b0b3e4fa/src/plugins/region_map/public/region_map_visualization.js#L190

```
   ...
   this._choroplethLayer = new ChoroplethLayer(
          name,
          attribution,
          this._params.selectedLayer.format,
          showAllData,
          this._params.selectedLayer.meta,
          this._params.selectedLayer,
          this._serviceSettings,
          (await lazyLoadMapsLegacyModules()).L
        );
   ....
```

And then rename to something more explicit

```

...

export class GeohashLayer extends KibanaMapLayer {
  constructor(featureCollection, featureCollectionMetaData, options, zoom, kibanaMap, leafletModule) {
    super();

   this._leafletModule = leafletModule;

...
```

Similar for `_serviceSettings`